lte/gateway/python/magma/pipelined/tests/test_conntrack.py

```diff
@@ -0,0 +1,128 @@
+"""
+Copyright (c) 2019-present, Facebook, Inc.
+All rights reserved.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree. An additional grant
+of patent rights can be found in the PATENTS file in the same directory.
+"""
+
+import unittest
+import warnings
+from concurrent.futures import Future
+
+from lte.protos.mconfig.mconfigs_pb2 import PipelineD
+from magma.pipelined.app.conntrack import ConntrackController
+from magma.pipelined.tests.app.start_pipelined import TestSetup, \
+    PipelinedController
+from magma.pipelined.tests.app.packet_builder import TCPPacketBuilder
+from magma.pipelined.tests.app.subscriber import SubContextConfig
+from magma.pipelined.tests.app.table_isolation import RyuDirectTableIsolator, \
+    RyuForwardFlowArgsBuilder
+from magma.pipelined.tests.app.packet_injector import ScapyPacketInjector
+from magma.pipelined.bridge_util import BridgeTools
+from magma.pipelined.tests.pipelined_test_util import start_ryu_app_thread, \
+    stop_ryu_app_thread, create_service_manager, assert_bridge_snapshot_match, \
+    SnapshotVerifier
+from ryu.lib import hub
+
+class ConntrackTest(unittest.TestCase):
+    BRIDGE = 'testing_br'
+    IFACE = 'testing_br'
+    MAC_DEST = "5e:cc:cc:b1:49:4b"
+    BRIDGE_IP = '192.168.128.1'
+    INBOUND_TEST_IP = '127.0.0.1'
+    OUTBOUND_TEST_IP = '127.1.0.1'
+    BOTH_DIR_TEST_IP = '127.2.0.1'
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Starts the thread which launches ryu apps
+
+        Create a testing bridge, add a port, setup the port interfaces. Then
+        launch the ryu apps for testing pipelined. Gets the references
+        to apps launched by using futures.
+        """
+        super(ConntrackTest, cls).setUpClass()
+        warnings.simplefilter('ignore')
+        cls.service_manager = create_service_manager([],
+            ['ue_mac', 'conntrack'])
+        cls._tbl_num = cls.service_manager.get_table_num(
```

Comment:
why not add IPFIX APP and validate flow stats in IPFIX table.

Reply:
The default agw ovs doesn't have ipfix patches so it will need some modifications to work. thats why we don't have any tests for ipfix table

Reply:
lets fix it, Can you create task to build vagrant vm with the unified OVS build.